vpnmanager/vpnservicemanager/src/main/java/com/kape/vpnservicemanager/data/externals/Service.kt

```diff
@@ -41,6 +41,7 @@ internal class Service :
     private lateinit var protocol: IProtocol
     private lateinit var subnet: ISubnet
     private lateinit var cache: ICache
+    private var gateway: String? = null
```

Comment:
use the `cache`. You create the set/get/clearGateway and make sure to add the clear as part of the `clear()` method you'll find there. In general we don't want to add instance variables on the objects but rather define a cache interface (if needed) for what we want to do, and let the cache deal with states.

Reply:
Sure thing, will do.